rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -1047,3 +1059,87 @@ def test_taxable_acquisition(accountant):
     assert pot.processed_events[0].free_amount == ZERO
     assert pot.processed_events[0].pnl.taxable == ONE
     assert pot.processed_events[0].pnl.free == ZERO
+
+
+@pytest.mark.parametrize('mocked_price_queries', [{
+    A_ETH: {
+        A_EUR: {
+            1677593073: FVal(50),
+            1677593074: FVal(120),
+            1677593075: FVal(130),
+            1677593076: FVal(90),
+        },
+    },
+}])
+@pytest.mark.parametrize(('db_settings', 'expected_pnls'), [
+    (
+        {'cost_basis_method': CostBasisMethod.FIFO, 'include_fees_in_cost_basis': False},
+        [ZERO, ZERO, FVal(-10), FVal(3500), ZERO, FVal(-10), ZERO, ZERO, FVal(-10), FVal(1600), ZERO, FVal(-10)],  # noqa: E501
+    ),
+    (
+        {'cost_basis_method': CostBasisMethod.ACB, 'include_fees_in_cost_basis': True},
+        [ZERO, ZERO, FVal(3495), ZERO, ZERO, ZERO, FVal(-6), ZERO],
```

Comment:
Continuing the conversation from before. 

> costs should be subtracted when selling, but in fact transaction costs are always about
    buy cost basis 

This does not make sense. I think you mean you want to keep the fees to add them to the buy cost basis. This I am not sure about. Perhaps it should be configurable? Perhaps it's an overkill ... but in this example ... it shoudl not be part of the buy cost basis.

Since the "buy" part of the sell here is fiat. And for fiat we do not count cost basis. Fiat is the profit currency. So with what you wrote the 10$ fee is lost here. The results for this example should be the same as my comment issue that you linked, not with $10 more. It's also same as here: https://www.adjustedcostbase.ca/blog/how-to-calculate-adjusted-cost-base-acb-and-capital-gains/


Now .... there is a valid question about what to do if it was something like sell ETH for BTC. There .. I think we can go with the buy cost basis thing. So in short, I believe:
1. This example should match the issue you linked.
2. You need to configure the implementation to act differently depending on if the sell is for FIAT or not.
    a. If it's for FIAT, then you should add the fee as per this example
    b. If it's for something that also gets its own cost basis, then add it to its cost basis.

And maybe... (but let's leave that for later) this behaviour can also be customized.

Reply:
Makes sense